crates/prune/src/builder.rs

```diff
@@ -0,0 +1,80 @@
+use crate::{segments::SegmentSet, Pruner};
+use reth_db::database::Database;
+use reth_primitives::{PruneModes, MAINNET};
+use reth_provider::ProviderFactory;
+use reth_snapshot::HighestSnapshotsTracker;
+
+/// Contains the information required to build a pruner
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PrunerBuilder {
+    /// Minimum pruning interval measured in blocks.
+    pub block_interval: usize,
+    /// Pruning configuration for every part of the data that can be pruned.
+    pub segments: PruneModes,
+    /// The number of blocks that can be re-orged.
+    pub max_reorg_depth: usize,
+    /// The delete limit for pruner, per block. In the actual pruner run it will be multiplied by
+    /// the amount of blocks between pruner runs to account for the difference in amount of new
+    /// data coming in.
+    pub prune_delete_limit: usize,
+}
+
+impl PrunerBuilder {
+    /// Creates a new [PrunerBuilder] with default values.
+    pub fn new() -> Self {
```

Comment:
this function is redundant, I'd like to preserve new, in case we need it

Reply:
changed it to `new(config: PrunerConfig)`